packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -1183,10 +1184,8 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
     const providers: Set<AuthProvider> = new Set();
     // get the roles created for type
     roles.forEach((role) => providers.add(this.roleMap.get(role)!.provider));
-    if (this.configuredAuthProviders.hasAdminRolesEnabled) {
-      if (!providers.has('iam') && !providers.has('identityPool')) {
-        providers.add('identityPool');
-      }
+    if (this.configuredAuthProviders.hasAdminRolesEnabled || this.configuredAuthProviders.genericIamAccessEnabled) {
+      providers.add('identityPool');
```

Comment:
This was the only conflict in logic.
Resolved `if` condition in favor of `gen2-release` branch.
Resolved content of `if` in favor of `identityPool` branch.

Note. The fact that we "pretend to have iam provider" to force other logic to add service directive is pre-existing art applied for admin roles which just got extended to cover iam access and new provider name.